internal/cmd/onepasswordsdktemplatefuncs.go

```diff
@@ -0,0 +1,53 @@
+package cmd
+
+import (
+	"context"
+
+	"github.com/1password/onepassword-sdk-go"
+)
+
+type onepasswordSDKConfig struct {
+	Token        string `json:"token" mapstructure:"token" yaml:"token"`
+	resolveCache map[string]string
+	client       *onepassword.Client
+	clientErr    error
+}
+
+func (c *Config) onepasswordResolve(secretReference string) string {
+	if result, ok := c.OnepasswordSDK.resolveCache[secretReference]; ok {
+		return result
+	}
+
+	ctx := context.TODO()
+
+	client, err := c.onepasswordSDKClient(ctx)
+	if err != nil {
+		panic(err)
+	}
+
+	secret, err := client.Secrets.Resolve(ctx, secretReference)
+	if err != nil {
+		panic(err)
+	}
+
+	if c.OnepasswordSDK.resolveCache == nil {
+		c.OnepasswordSDK.resolveCache = make(map[string]string)
+	}
+	c.OnepasswordSDK.resolveCache[secretReference] = secret
+
+	return secret
+}
+
+func (c *Config) onepasswordSDKClient(ctx context.Context) (*onepassword.Client, error) {
+	if c.OnepasswordSDK.client != nil || c.OnepasswordSDK.clientErr != nil {
+		return c.OnepasswordSDK.client, c.OnepasswordSDK.clientErr
+	}
+
+	c.OnepasswordSDK.client, c.OnepasswordSDK.clientErr = onepassword.NewClient(
+		ctx,
+		onepassword.WithServiceAccountToken(c.OnepasswordSDK.Token),
+		onepassword.WithIntegrationInfo("chezmoi", c.versionInfo.Version),
```

Comment:
I'm getting the following error when retrieving a secret:

`chezmoi: template: arg1:1:3: executing "arg1" at <onepasswordResolve "op://ihxdr5a6susziu5yo5rlj4ypeq/Login/password">: error calling onepasswordResolve: error initializing client: invalid user input: encountered the following errors: integration version was not specified`

To verify, I changed `c.versionInfo.Version` to `"1.0"` and confirmed that it works as expected. I'm wondering if it could be related to my local build and the version number not being set.

Reply:
Thanks for the testing. You're right, the version needs to be specified when building, otherwise it defaults to the empty string.

Can you build chezmoi with:

```console
$ go build -ldflags "-X main.version=v2.0.0" .
```

and retry the test with the resulting binary?